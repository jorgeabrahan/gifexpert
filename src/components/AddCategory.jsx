import { useState } from 'react';

export const AddCategory = ({onAddCategory}) => {
    const [inputValue, setInputValue] = useState("")
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }
    const onSubmit = (e) => {
        e.preventDefault();
        const category = inputValue.trim(); 
        //Si no ingreso una categor√≠a
        if (category.length <= 1) return;
        
        onAddCategory(category); //Se manda a llamar una funcion que agregue la categoria al estado

        setInputValue(""); //Se vacia el input
    }
    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Busca una categoria"
                onChange={ onInputChange }
                value={inputValue}
            />
        </form>
    )
}
