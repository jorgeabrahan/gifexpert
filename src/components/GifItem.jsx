import PropTypes from 'prop-types'; //npm install prop-types *solo en vite*

export const GifItem = ({title, url}) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

/* Especificar el tipo de dato que se espera recibir en las propiedades y hacerla obligatoria */
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}