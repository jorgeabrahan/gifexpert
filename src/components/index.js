/* Este es llamado archivo de barril y esta destinado a exportar los archivos de una carpeta */
export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";