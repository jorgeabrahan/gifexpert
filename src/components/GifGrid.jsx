import PropTypes from 'prop-types';
//Importaciones de componentes de terceros
import { GifItem } from "./GifItem";
//Importaciones de funciones u otras cosas que no sean componentes
import { useFetchGifs } from "../hooks/useFetchGifs";



export const GifGrid = ({category}) => {

    const { images, isLoading } = useFetchGifs(category);
    
    return (
        <div key={category}>
            <h3>{category}</h3>
            
            {isLoading && <h2>Cargando...</h2>}

            <div className="card-grid">
                {
                    images.map(image => (
                        <GifItem
                            key={image.id}
                            {...image} //Se esparcen las propiedades de un objeto para que puedan ser accedidas como props
                        />
                    ))
                }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}