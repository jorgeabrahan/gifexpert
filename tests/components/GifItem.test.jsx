import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe("Pruebas al componente GifItem", () => {
    const title = "Titulo de prueba";
    const url = "https://urldeprueba.com/";

    test("Debe hacer match con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test("Debe mostrar la imagen con el URL y el Alt indicado", () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug(); permite mostar el screen
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

})