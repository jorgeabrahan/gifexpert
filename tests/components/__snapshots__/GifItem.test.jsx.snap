// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de prueba"
      src="https://urldeprueba.com/"
    />
    <p>
      Titulo de prueba
    </p>
  </div>
</div>
`;
